### ISSUE TITLE
Docker Swarm Support

### ISSUE THREAD

**Issue by iLuckyW** [mbecker20/monitor#37]
Is there a plan to support docker swarm? Would be really interested to replace portainer with monitor as a tool to monitor a docker swarm.

**Comment by mbecker20** [comment]
Good point, yes I think this would be very nice, I'll add it to the roadmap near Kubernetes support.

**Comment by mrvalterhugo** [comment]
This would be very nice to have, I run a swarm cluster in my proxmox server, using LXC containers.
Portainer supports it and it is useful.

**Comment by Surrogard** [comment]
I would be very interested in this as well. I have a 12 node Docker swarm, so portainer is out for me. 

If you need help with anything, just ping me.

**Comment by mikemilligram** [comment]
Also interested in this, running swarm across four nodes. I'm by no means an expert but I'd love to help out in any way possible!

**Comment by TC74** [comment]
Docker swarm support is a good (must) feature to have, as there is no alternative out there other than Portainer. You can make it free, and sell entreprise support to fund this project. 

IMO, focus on docker stack and swarm will make komo.do the go to solution for docker goer that is looking for simple setup. Swarmpit was the go to, but the project is now dead. Rancher stopped supporting Docker Swarm. Portainer basically need to purchase licenses (I was willing to pay, but does not like the license based charging model... hahaha). 

Looking forward to Docker Stack and Swarm support.

**Comment by StefanSa** [comment]
@mbecker20 
Supporting Docker Swarm is a crucial feature for Komodo, as it fills a gap in the market with no real alternatives besides Portainer.
By integrating Docker Swarm support, Komodo can become the go-to solution for Docker users seeking a simple setup

**Comment by Sedation6612** [comment]
any ETA on swarm support? can't wait to swap once it's added

**Comment by nikhilmaddirala** [comment]
> any ETA on swarm support? can't wait to swap once it's added

It was previously listed on the roadmap for the next version, but now it's moved to "undecided".

**Comment by Kieranwest** [comment]
Would love Docker Swarm support. I'm trying my best to avoid portainer but it seems to be the only option to manage swarms nicely. I guess the only option right now is using the Sync feature of Komodo to replicate the stacks.toml file so i can spin up containers on any node in the event of an outage :(. 
Pretty please with a cherry on-top we'd love swarm support <3

**Comment by mbecker20** [comment]
I'm putting it off to solidify the base before adding more features. There have continuously been other pressing features and issues that must be addressed before continuing to expand the scope.

**Comment by djakupovic** [comment]
I currently use Komodo with five agents on a 5 node docker swarm cluster as "external" docker servers. This is my main use case for Komodo to deploy "docker compose up" command on specific nodes. 

If a swarm feature comes, please give us an option to have docker stack deploy and docker compose up.

Great work so far and thank you for your amazing tool

**Comment by Kieranwest** [comment]
> I currently use Komodo with five agents on a 5 node docker swarm cluster as "external" docker servers. This is my main use case for Komodo to deploy "docker compose up" command on specific nodes.

So essentially, you're hosting a Komodo Control Server and then pointing to an external periphery that lives on Docker swarm? If that's the case how does periphery handle creating the Docker containers? Does it run then underneath swarm?

**Comment by djakupovic** [comment]
> > I currently use Komodo with five agents on a 5 node docker swarm cluster as "external" docker servers. This is my main use case for Komodo to deploy "docker compose up" command on specific nodes.
> 
> So essentially, you're hosting a Komodo Control Server and then pointing to an external periphery that lives on Docker swarm? If that's the case how does periphery handle creating the Docker containers? Does it run then underneath swarm?

Yes thats how I have done it. Yes it applies docker compose up I assume instead docker stack deploy. This is why I use Komodo. I use portioner for my docker stack - swarm containers - and Komodo for lokal deployments on specific nodes e.g. docker 1 and docker 4 

I would like to replace Portainer with Komodo if it has a swarm feature so I need to wait until it is finished. Then portainer is deprecated for me

**Comment by mattsmallman** [comment]
Just adding a plus 1 for swarm support (when the time is right). I’m a huge fan of komodo and have been trying to get blue/green deployments working for my laravel docker apps but it’s just to messy with multiple stacks and scripts so would prefer to use a docker swarm but don’t want to give up the Komodo interface.

**Comment by Gsyltc** [comment]
And One more plus.
Dowker swarm support will be a really good feature. Espacally for homelab

**Comment by CodeShellDev** [comment]
Love Komodo sofar would really like to try out Docker Swarm with it. 👍

**Comment by foureight84** [comment]
Another vote for Docker Swarm support

**Comment by Deester4x4jr** [comment]
👍 +1 for this as well. Was going to deploy a container on a test machine with automated caddy reverse proxy, but without overlay networks its a no go. Admittedly a pretty shallow use case, but still would love to have it.

**Comment by syphernl** [comment]
> 👍 +1 for this as well. Was going to deploy a container on a test machine with automated caddy reverse proxy, but without overlay networks its a no go. Admittedly a pretty shallow use case, but still would love to have it.

FWIW you already can create swarms and overlay networks by hand (or something like Ansible) and use them within Komodo. 
The only thing to keep in mind is to specify this network within your compose files e.g.:

```yaml

services:
  foo:
    # Your service config here
    networks:
      overlay_network:
        aliases:
          - myservice.svc

networks:
  overlay_network:
    name: overlay_network
    external: true
```

I have been using this setup for a couple of months now and it is working fine. My containers can communicate with each other over the overlay network across all Docker nodes.
The only thing I noticed is that while Komodo does show the network in the UI clicking it only shows the containers & IP's connected to it from the same node, not on the other nodes.

**Comment by TheLinuxGuy** [comment]
This feature would be a great addition to komodo, hope it lands soon.

I have been playing with Docker Swarm on 3 nodes with dokploy.com and there are 2 features from dokploy that I wish komodo would get from it: 
- Docker Swarm
- Integrated / automated backups (for [databases](https://docs.dokploy.com/docs/core/databases/backups) and [volumes](https://docs.dokploy.com/docs/core/volume-backups))

**Comment by rkbks** [comment]
We're planning to move from standalone docker hosts to a swarm setup. Support for docker swarm would be really awesome. We looked at alternatives like Portainer but we don't like how things are going there. Komodo is way superior! Keep up the good work guys.

**Comment by CodeShellDev** [comment]
This is for Docker Swarm MODE right? Not classic?
Just wanted to make sure...

**Comment by acheslow** [comment]
+1 swarm support is the only thing keeping me from moving from Portainer

**Comment by maradanasai** [comment]
+1 swarm support is the only thing keeping me as well like others, to move from portainer

**Comment by epoch-philosophy** [comment]
+1 Docker swarm support would be huge. And would move me over to Komodo. Just adding my voice here too!

**Comment by waynehaffenden** [comment]
+1 for Docker swarm support here too!

**Comment by KyleMercer** [comment]
+1 for Docker swarm support. Would love to move from Portainer!

**Comment by foureight84** [comment]
> This is for Docker Swarm MODE right? Not classic? Just wanted to make sure...

Yes, Docker Swarm Mode.

**Comment by NuroDev** [comment]
Looking at #889 of what's to come for v2 of Komodo, it looks like Swarm support is currently being worked on, or at least experimented with 🎉 

Cannot wait for this @mbecker20!

**Comment by mbecker20** [comment]
Initial support will come with v2 and be included in the next dev release. It's not complete yet
